.github/workflows/release.yaml

```diff
@@ -9,7 +9,12 @@ jobs:
     - uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 # v3.5.0
       with:
         go-version: "1.19.6"
-    - run: |
+    - name: test-image
+      run: |
+        make test-images
+        make clean
+    - name: release
```

Comment:
The step `test-image` should be put after `release`, and uses the images generated by `release` directly.

Reply:
True, however we should test image before we release it. So flow should be
```
make release
make test-release
make publish-release
```

Reply:
Makes sense to me. 

Please try to breakdown the task, either into two separate PRs (preferred) or a couple of small commits in this PR. @pchan 
- breakdown the existing `release` into two targets such as `create-release` and `publish-release`;
- add a new target `test-release` in between to verify the image.

Reply:
@pchan Please let me know if you need help on the first task (breakdown the existing release).

Reply:
That implies we need to refactor release.sh, maybe modularize it or use separate scripts for `create-release` and `publish-release`. Can we use this PR to just track the addition of tests, I will create a separate issue and PRs to address those ? 

One stop-gap fix, till we fix the above, is to add a release-dir to the release script and test images so that we can re-use it. I have attempted it in the latest commit.

Reply:
> Can we use this PR to just track the addition of tests

It's OK to me. It's always recommended to do things step by step.

Reply:
I have created issue https://github.com/etcd-io/etcd/issues/15396 to track the refactoring. As an interim fix, I have changed the logic to re-use the artefacts of the release workflow.